x-pack/plugins/apm/e2e/ingest-data/replay.js

```diff
@@ -70,34 +70,40 @@ function incrementSpinnerCount({ success }) {
 }
 let iterIndex = 0;
 
+function setItemMetaAndHeaders(item) {
+  const headers = {
+    'content-type': 'application/x-ndjson',
+  };
+
+  if (SECRET_TOKEN) {
+    headers.Authorization = `Bearer ${SECRET_TOKEN}`;
+  }
+
+  if (item.url === '/intake/v2/rum/events') {
+    if (iterIndex === userAgents.length) {
+      // set some event agent to opbean
+      setRumAgent(item);
+      iterIndex = 0;
+    }
+    headers['User-Agent'] = userAgents[iterIndex];
+    headers['X-Forwarded-For'] = userIps[iterIndex];
+    iterIndex++;
+  }
+  return headers;
+}
+
```

Comment:
IMO we ran into this situation because of side-effects. If we can make this (more) functional, i.e. don't mutate `item` and `iterIndex`, I feel like this code would be more robust and easier to understand. I think changing `insertItem()` to something like `createRequestFromEvent()` would help.

Reply:
Sure, i will do it as a follow up PR, want to get this in to make sure it's working on other PRs.